### PR TITLE
fix(go): return errors correctly from Go client

### PR DESCRIPTION
**Pull request by markphelps**
Trying to test invalid authentication errors from #347

### CODE REVIEW

File: flipt-client-go/evaluation_test.go [hunk 1]
```diff
@@ -30,6 +34,11 @@ func init() {
 	}
 }
 
+func TestInvalidAuthentication(t *testing.T) {
+	_, err := flipt.NewClient(flipt.WithURL(fliptUrl), flipt.WithClientTokenAuthentication("invalid"))
+	assert.EqualError(t, err, "invalid authentication")
```
Comment (markphelps) [inline flipt-client-go/evaluation_test.go:39]: i know this isnt the correct error string, but i just wanted to see if there is any error

Comment (erka) [reply flipt-client-go/evaluation_test.go:39]: we need to do some `variant, err := evaluationClient.EvaluateVariant(...)` to get the error.

Comment (markphelps) [reply flipt-client-go/evaluation_test.go:43]: it should return the error on the initial pulling of the state I would think. I updated the test and still no error, maybe we arent returning it correctly from the engine?

Comment (erka) [reply flipt-client-go/evaluation_test.go]: let me give it a try.

**Review by erka — APPROVED** [review]
nice. I believe this actually the fix for original issue. ty!

**Pull request merged**